core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -165,12 +165,24 @@ public static RecordState forId(byte id) {
      */
     private final int maxDeliveryCount;
 
+    /**
+     * recordLockPartitionLimit stores the Share-group record lock limit per share-partition. This value is retrieved
+     * from the config group.share.record.lock.partition.limit
+     */
+    private final short recordLockPartitionLimit;
```

Comment:
I am wondering what the difference is between maxInFlightMessages and recordLockPartitionLimit. I notice that you're using 100 to initialize maxInFlightMessages and then the config to initialize recordLockPartitionLimit. It seems to me that they are really the same concept and you can use the config to initialize maxInFlightMessages.